### PR TITLE
84348: Create job DeleteSavedClaimRecordsJob 

### PR DESCRIPTION
**Pull request by dfong-adh**
## Summary
This adds a scheduled recurring job to delete records from `SavedClaims` with a `delete_date` timestamp in the past. We will delete data we no longer need in accordance with our ATO.

This is behind the flipper flag `decision_review_delete_saved_claims_job_enabled`

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/84348

## Testing done

- [x] *New code is covered by unit tests*
Locally tested + respec tests

## What areas of the site does it impact?
This affects the SavedClaims table.

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by shaunburdick — DISMISSED** [review]
`LGTM.destroy_all`

**Comment by rmtolmach** [comment]
your unit tests were failing. I am rerunning them.

File: app/sidekiq/decision_review/delete_saved_claim_records_job.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  class DeleteSavedClaimRecordsJob
+    include Sidekiq::Job
+
+    # No need to retry since the schedule will run this periodically
+    sidekiq_options retry: false
+
+    def perform
+      return unless enabled?
+
+      ::SavedClaim.where(delete_date: ..DateTime.now).destroy_all
```
Comment (rmtolmach) [inline app/sidekiq/decision_review/delete_saved_claim_records_job.rb:15]: Do you anticipate the possibility of errors? should there be a rescue and a Rails.logger.error?

Comment (dfong-adh) [reply app/sidekiq/decision_review/delete_saved_claim_records_job.rb:15]: I'll add a rescue and log the error for any exceptions.

**Pull request merged**